shell.js

```diff
@@ -42,69 +44,92 @@ process.argv.forEach(function(val, index, array) {
 });
 
 if (opts.brain === undefined) {
-	console.log("Usage: node shell.js [--debug --utf8] </path/to/brain>");
+	console.log("Usage: node shell.js [--debug --utf8 --watch] </path/to/brain>");
 	process.exit(1);
 }
 
 //------------------------------------------------------------------------------
 // Initialize the RiveScript bot and print the welcome banner.
 //------------------------------------------------------------------------------
 
-var bot = new RiveScript({
-	debug: opts.debug,
-	utf8:  opts.utf8,
-});
-bot.loadDirectory(opts.brain, loading_done, loading_error);
-
-function loading_done(batch_num) {
-	bot.sortReplies();
+var bot = null;
+function loadBot() {
+	bot = new RiveScript({
+		debug: opts.debug,
+		utf8:  opts.utf8,
+	});
+	bot.ready = false;
+	bot.loadDirectory(opts.brain, loading_done, loading_error);
+}
+loadBot();
 
-	console.log("      .   .       ");
-	console.log("     .:...::      RiveScript Interpreter (JavaScript)");
-	console.log("    .::   ::.     Library Version: v" + bot.version());
-	console.log(" ..:;;. ' .;;:..  ");
-	console.log("    .  '''  .     Type '/quit' to quit.");
-	console.log("     :;,:,;:      Type '/help' for more options.");
-	console.log("     :     :      ");
-	console.log("");
-	console.log("Using the RiveScript bot found in: " + opts.brain);
-	console.log("Type a message to the bot and press Return to send it.");
-	console.log("");
+if (opts.watch) {
+	fs.watch(opts.brain, {recursive: false}, function() {
+		console.log("");
+		console.log('[INFO] Brain changed, reloading bot.');
+		rl.prompt();
+		loadBot();
+	});
+}
 
-	//--------------------------------------------------------------------------
-	// Drop into the interactive command shell.
-	//--------------------------------------------------------------------------
 
-	var rl = readline.createInterface({
-		input: process.stdin,
-		output: process.stdout
-	});
+//--------------------------------------------------------------------------
+// Drop into the interactive command shell.
+//--------------------------------------------------------------------------
+
+console.log("      .   .       ");
+console.log("     .:...::      RiveScript Interpreter (JavaScript)");
+console.log("    .::   ::.     Library Version: v" + bot.version());
+console.log(" ..:;;. ' .;;:..  ");
+console.log("    .  '''  .     Type '/quit' to quit.");
+console.log("     :;,:,;:      Type '/help' for more options.");
+console.log("     :     :      ");
+console.log("");
+console.log("Using the RiveScript bot found in: " + opts.brain);
+console.log("Type a message to the bot and press Return to send it.");
+console.log("");
+
+var rl = readline.createInterface({
+	input: process.stdin,
+	output: process.stdout
+});
```

Comment:
I know JavaScript hoists variable declarations, but this variable was referred to on line 70 and only declared down here which looks a little weird (disregard if you end up removing the line 70).